[Profiler][Trivial] Cleanup ExperimentalConfig

robieta opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #83965
* #83964
* #83895
* #83894
* #83893
* #83892
* #83891
* __->__ #83890
* #83963
* #83622

I'm trying to limit how much is in headers to make it easier to read the API surface. In a similar vein, we can replace `hasOptions` with `operator bool` so it just does the right thing in the check.

Differential Revision: [D38917366](https://our.internmc.facebook.com/intern/diff/D38917366/)

@diff-train-skip-merge

Commit:
[Profiler][Trivial] Cleanup ExperimentalConfig

I'm trying to limit how much is in headers to make it easier to read the API surface. In a similar vein, we can replace `hasOptions` with `operator bool` so it just does the right thing in the check.

Differential Revision: [D38917366](https://our.internmc.facebook.com/intern/diff/D38917366/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Trivial] Cleanup ExperimentalConfig"


I'm trying to limit how much is in headers to make it easier to read the API surface. In a similar vein, we can replace `hasOptions` with `operator bool` so it just does the right thing in the check.

Differential Revision: [D38917366](https://our.internmc.facebook.com/intern/diff/D38917366/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Trivial] Cleanup ExperimentalConfig"


I'm trying to limit how much is in headers to make it easier to read the API surface. In a similar vein, we can replace `hasOptions` with `operator bool` so it just does the right thing in the check.

Differential Revision: [D38917366](https://our.internmc.facebook.com/intern/diff/D38917366/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Trivial] Cleanup ExperimentalConfig"


I'm trying to limit how much is in headers to make it easier to read the API surface. In a similar vein, we can replace `hasOptions` with `operator bool` so it just does the right thing in the check.

Differential Revision: [D38917366](https://our.internmc.facebook.com/intern/diff/D38917366/)

[ghstack-poisoned]

slgong-fb commented on torch/csrc/profiler/api.cpp:
```diff
@@ -4,6 +4,16 @@ namespace torch {
+    : profiler_metrics{profiler_metrics},
+      profiler_measure_per_kernel{profiler_measure_per_kernel} {}
+
+ExperimentalConfig::operator bool() const {
```
`explicit operator bool()` would not be a safer way to do this?

robieta replied:
That's a great point.

Commit:
Update on "[Profiler][Trivial] Cleanup ExperimentalConfig"


I'm trying to limit how much is in headers to make it easier to read the API surface. In a similar vein, we can replace `hasOptions` with `operator bool` so it just does the right thing in the check.

Differential Revision: [D38917366](https://our.internmc.facebook.com/intern/diff/D38917366/)

[ghstack-poisoned]

weiwangmeta commented:
Not understanding why/what is causing a revert again.
